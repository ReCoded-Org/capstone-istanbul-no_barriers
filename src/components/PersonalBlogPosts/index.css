.personalBlogPostsBtn,
.personalBlogPostsBtn:focus {
  border: none;
  background-color: var(--main-text-color-profile);
  color: var(--blog-background-profile);
}

/* write blog post button style on click */
.personalBlogPostsBtn:hover,
.personalBlogPostsBtn.btn-primary:not(:disabled):not(.disabled):active {
  background-color: var(--main-title-profile);
  color: var(--main-background-color-profile);
  border: none;
}
