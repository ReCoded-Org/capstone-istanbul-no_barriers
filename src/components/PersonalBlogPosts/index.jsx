import React from "react";
import { Row, Button } from "react-bootstrap";
import RenderBlogPost from "../RenderBlogPost";
import { useTranslation } from "react-i18next";
import "./index.css";
import postBackground from "./profile.jpg";

const PersonalBlogPosts = () => {
  const { t } = useTranslation();
  const recentPosts = [
    { id: "aa", image: postBackground, title: "Title", date: "11/11/11" },
    { id: "bb", image: postBackground, title: "Title", date: "11/11/11" },
    { id: "ba", image: postBackground, title: "Title", date: "11/11/11" },
  ];

  return (
    <div className="m-5 p-5">
      <Row>
        <Button className="personalBlogPostsBtn shadow-none mb-1">
          {t("profile.writePost")}
        </Button>
      </Row>
      <Row>
        <h5>{t("profile.recentPost")}</h5>
      </Row>
      <Row className="justify-content-between">
        {recentPosts.map((post) => (
          <RenderBlogPost key={post.id} postInfo={post} />
        ))}
      </Row>
    </div>
  );
};

export default PersonalBlogPosts;
