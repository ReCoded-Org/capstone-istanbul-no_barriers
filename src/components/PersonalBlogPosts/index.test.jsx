import React from "react";
import renderer from "react-test-renderer";
import PersonalBlogPosts from "./index";

it("PersonalBlogPosts renders correctly", () => {
  const tree = renderer.create(<PersonalBlogPosts />).toJSON();
  expect(tree).toMatchSnapshot();
});
