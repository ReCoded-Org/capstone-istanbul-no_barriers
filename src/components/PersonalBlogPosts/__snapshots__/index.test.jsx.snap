// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalBlogPosts renders correctly 1`] = `
<div
  className="m-5 p-5"
>
  <div
    className="row"
  >
    <button
      className="personalBlogPostsBtn shadow-none mb-1 btn btn-primary"
      disabled={false}
      type="button"
    >
      profile.writePost
    </button>
  </div>
  <div
    className="row"
  >
    <h5>
      profile.recentPost
    </h5>
  </div>
  <div
    className="justify-content-between row"
  >
    <div
      className="pt-2 profileBlogPost"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="postInfo"
        className="profileBlogPostImage img-fluid"
        src="profile.jpg"
      />
      <div
        className="profileBlogPostInfo px-2 pt-2"
      >
        <h5>
          Title
        </h5>
        <p
          className="pb-2"
        >
          11/11/11
        </p>
      </div>
    </div>
    <div
      className="pt-2 profileBlogPost"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="postInfo"
        className="profileBlogPostImage img-fluid"
        src="profile.jpg"
      />
      <div
        className="profileBlogPostInfo px-2 pt-2"
      >
        <h5>
          Title
        </h5>
        <p
          className="pb-2"
        >
          11/11/11
        </p>
      </div>
    </div>
    <div
      className="pt-2 profileBlogPost"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="postInfo"
        className="profileBlogPostImage img-fluid"
        src="profile.jpg"
      />
      <div
        className="profileBlogPostInfo px-2 pt-2"
      >
        <h5>
          Title
        </h5>
        <p
          className="pb-2"
        >
          11/11/11
        </p>
      </div>
    </div>
  </div>
</div>
`;
